source/patterns/@aws-solutions-constructs/aws-fargate-sns/lib/index.ts

```diff
@@ -0,0 +1,182 @@
+  /**
+   * The version of the image to use from the repository
+   *
+   * @default - 'latest'
+   */
+  readonly ecrImageVersion?: string;
+  /*
+   * Optional props to define the container created for the Fargate Service
+   *
+   * defaults - fargate-defaults.ts
+   */
+  readonly containerDefinitionProps?: ecs.ContainerDefinitionProps | any;
+  /*
+   * Optional props to define the Fargate Task Definition for this construct
+   *
+   * defaults - fargate-defaults.ts
+   */
+  readonly fargateTaskDefinitionProps?: ecs.FargateTaskDefinitionProps | any;
+  /**
+   * Optional values to override default Fargate Task definition properties
+   * (fargate-defaults.ts). The construct will default to launching the service
+   * is the most isolated subnets available (precedence: Isolated, Private and
+   * Public). Override those and other defaults here.
+   *
+   * defaults - fargate-defaults.ts
+   */
+  readonly fargateServiceProps?: ecs.FargateServiceProps | any;
+  /**
+   * A Fargate Service already instantiated (probably by another Solutions Construct). If
+   * this is specified, then no props defining a new service can be provided, including:
+   * existingImageObject, ecrImageVersion, containerDefintionProps, fargateTaskDefinitionProps,
+   * ecrRepositoryArn, fargateServiceProps, clusterProps, existingClusterInterface. If this value
+   * is provided, then existingContainerDefinitionObject must be provided as well.
+   *
+   * @default - none
+   */
+  readonly existingFargateServiceObject?: ecs.FargateService;
+  /**
+   * Existing instance of SNS Topic object, providing both this and topicProps will cause an error..
+   *
+   * @default - Default props are used
+   */
+  readonly existingTopicObject?: sns.Topic;
+  /**
+   * Optional user provided properties to override the default properties for the SNS topic.
+   *
+   * @default - Default properties are used.
+   */
+  readonly topicProps?: sns.TopicProps;
+  /**
+   * Optional Name for the SNS topic arn environment variable set for the Lambda function.
+   *
+   * @default - None
+   */
+  readonly topicArnEnvironmentVariableName?: string;
+  /**
+   * Optional Name for the SNS topic name environment variable set for the Lambda function.
+   *
+   * @default - None
+   */
+  readonly topicNameEnvironmentVariableName?: string;
+  /*
+   * A container definition already instantiated as part of a Fargate service. This must
+   * be the container in the existingFargateServiceObject.
+   *
+   * @default - None
+   */
+  readonly existingContainerDefinitionObject?: ecs.ContainerDefinition;
+}
+
+export class FargateToSns extends Construct {
+  public readonly snsTopic: sns.Topic;
+  public readonly service: ecs.FargateService;
+  public readonly vpc: ec2.IVpc;
+  public readonly container: ecs.ContainerDefinition;
+
+  constructor(scope: Construct, id: string, props: FargateToSnsProps) {
+    super(scope, id);
+    defaults.CheckProps(props);
+    defaults.CheckFargateProps(props);
```

Comment:
CheckProps checks values that are spread across many constructs (e.g. - VPC). With ALB we had many checks unique to ALB that will apply only to ALB based constructs, so we broke those out into a function specifically for ALB constructs. Fargate was similar so we repeated that pattern. I think we'll probably move towards a check function for each primary service in a construct, plus the overall check. Right now SNS checks are in CheckProps - we didn't move them because a) there's lots of SNS constructs; b) there's only 1 or 2 SNS based checks. Fargate has 7, and a couple of them are more complicated than what's in CheckProps().